financial-templates-lib/GasEstimator.js

```diff
@@ -0,0 +1,71 @@
+// This script gets the current recommended `fast` gas price from ethGasStation
+// to inform the Liquidator and dispute bot of a reasonable gas price to use.
+
+const fetch = require("node-fetch");
+const url = "https://ethgasstation.info/json/ethgasAPI.json";
+const { Logger } = require("./Logger");
+
+// If no updateThreshold is specified then default to updating every 60 seconds.
+class GasEstimator {
+  constructor(updateThreshold = 60) {
+    this.updateThreshold = updateThreshold;
+    this.lastUpdateTimestamp;
+    this.lastFastPriceGwei;
+
+    // If the script fails or the API resonce fails default to this value
+    this.defaultFastPriceGwei = 40;
+  }
+
+  _update = async () => {
+    const currentTime = Math.floor(Date.now() / 1000);
+    if (currentTime < this.lastUpdateTimestamp + this.updateThreshold) {
+      Logger.debug({
+        at: "GasEstimator",
+        message: "Gas Estimator update skipped due to update threshold",
+        currentTime: currentTime,
+        lastUpdateTimestamp: this.lastUpdateTimestamp,
+        lastFastPriceGwei: this.lastFastPriceGwei,
+        timeRemainingUntilUpdate: this.lastUpdateTimestamp + this.updateThreshold - currentTime
+      });
+      return;
+    } else {
+      let returnedPrice = await this.getPrice(url);
+      this.lastFastPriceGwei = returnedPrice;
+      this.lastUpdateTimestamp = currentTime;
+      Logger.debug({
+        at: "GasEstimator",
+        message: "Gas Estimator updated",
+        currentFastPrice: returnedPrice,
+        lastUpdateTimestamp: this.lastUpdateTimestamp
+      });
+    }
+  };
+
+  // Returns the current fast gas price in Wei, converted from the stored Gwei value.
+  getCurrentFastPrice = () => {
+    return this.lastFastPriceGwei * 1e9;
```

Comment:
What happens if someone calls `getCurrentFastPrice()` right after `GasEstimator` construction? It will return null or 0. Should we set `this.lastFastPriceGwei = this.defaultFastPriceGwei` in the constructor`?

Reply:
ye, this is also an option. will set this now.